packages/kbn-pm/src/commands/reset.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import dedent from 'dedent';
+import del from 'del';
+import ora from 'ora';
+import { join, relative } from 'path';
+
+import { getBazelDiskCacheFolder, getBazelRepositoryCacheFolder, runBazel } from '../utils/bazel';
+import { isDirectory } from '../utils/fs';
+import { log } from '../utils/log';
+import { ICommand } from './';
+
+export const ResetCommand: ICommand = {
+  description:
+    'Deletes node_modules and output directories, resets internal and disk caches, and stops Bazel server',
+  name: 'reset',
+
+  async run(projects) {
+    log.warning(dedent`
+      In most cases, 'yarn kbn clean' is all that should be needed to recover a consistent state when
+      problems arise. If you need to use this command, please let us know, as it should not be necessary.
+    `);
+
+    const toDelete = [];
+    for (const project of projects.values()) {
+      if (await isDirectory(project.nodeModulesLocation)) {
+        toDelete.push({
+          cwd: project.path,
+          pattern: relative(project.path, project.nodeModulesLocation),
+        });
+      }
+
+      if (await isDirectory(project.targetLocation)) {
+        toDelete.push({
+          cwd: project.path,
+          pattern: relative(project.path, project.targetLocation),
+        });
+      }
+
+      const { extraPatterns } = project.getCleanConfig();
+      if (extraPatterns) {
+        toDelete.push({
+          cwd: project.path,
+          pattern: extraPatterns,
+        });
+      }
+    }
+
+    // Runs Bazel hard clean
+    await runBazel(['clean', '--expunge']);
+    log.success('Hard cleaned bazel');
+
+    // Deletes Bazel Cache Folders
+    await del([await getBazelDiskCacheFolder(), await getBazelRepositoryCacheFolder()], {
```

Comment:
Does expunge not do this already?

Reply:
I thought it did it but it don't. I realised that this afternoon.